PyRoute/Star.py

```diff
@@ -155,13 +155,25 @@ def __getstate__(self):
             del state['ownedBy']
         return state
 
+    def __deepcopy__(self, memodict={}):
+        state = self.__dict__.copy()
+
+        foo = Star()
+        for key in state:
+            item = state[key]
+            setattr(foo, key, item)
+
+        return foo
+
     @staticmethod
     def parse_line_into_star(line, sector, pop_code, ru_calc):
         star = Star()
         star.sector = sector
         star.logger.debug(line)
-        if Star.starline.match(line):
-            data = Star.starline.match(line).groups()
+        # Cache regex lookup to avoid doing it once for check, and again to extract data
```

Comment:
The duplicated regex lookups took up something like 95% of total runtime of this method, so this was an easy gain to collect.